src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -680,9 +680,13 @@ impl Step for Miri {
             .arg("--manifest-path")
             .arg(builder.src.join("src/tools/miri/test-cargo-miri/Cargo.toml"));
         cargo.arg("--target").arg(target.rustc_target_arg());
-        cargo.arg("--tests"); // don't run doctests, they are too confused by the staging
         cargo.arg("--").args(builder.config.test_args());
 
+        // `prepare_tool_cargo` sets RUSTDOC to the bootstrap wrapper and RUSTDOC_REAL to a dummy path as this is a "run", not a "test".
+        // Also, we want the rustdoc from the "next" stage for the same reason that we build a std from the next stage.
+        // So let's just set that here.
+        cargo.env("RUSTDOC", builder.rustdoc(compiler_std));
```

Comment:
I would expect this to still pass RUSTDOC_REAL, and use the bootstrap RUSTDOC wrapper.

Reply:
But I don't know why that wrapper is needed. So if you think it's better to bypass the wrapper, that works for me as well. After all we do also bypass the wrapper for RUSTC so it's kind of consistent.

Reply:
I don't have strong opinions on this. If you want I can pass the bootstrap's rustdoc wrapper. I just thought we don't need to use wrappers for this usecase.